src/chunk.c

```diff
@@ -716,10 +742,35 @@ ts_chunk_create(Hypertable *ht, Point *p, const char *schema, const char *prefix
 	LockRelationOid(ht->main_table_relid, ShareUpdateExclusiveLock);
 
 	/* Recheck if someone else created the chunk before we got the table lock */
-	chunk = ts_chunk_find(ht->space, p);
+	chunk = ts_chunk_find(ht->space, p, true);
+
+	if (chunk != NULL && chunk->fd.dropped)
+	{
+		/* mark as not dropped and recreate the pg objects */
+		ScanIterator iterator =
+			ts_scan_iterator_create(CHUNK, RowExclusiveLock, CurrentMemoryContext);
```

Comment:
when is the lock on the row released? we should be holding it till the pg object creation is complete.

Reply:
should we add an assert that the chunk is not a compressed chunk? Or should it be applicable to compressed chunks as well?

Reply:
we should test this scenario:  1) drop chunk on table 2) change hypertable's chunk interval 3) create some new chunks 4) now try to recreate the old chunk and insert/update/delete/select from the old chunk that was recreated with the old settings.

Reply:
The lock on line 716 is not released until end of txn. (see comment above 716). This should work on compressed chunks too. I have a test for this but will add the chunk_interval change.

Reply:
oh wait no it shouldn't be allowed on compressed chunks. I'll fix.

Reply:
Oh the compressed chunk thing is checked during drop.

Reply:
referring to the lock on line 751. RowExclusiveLock on CHUNK. when does that get released?

Reply:
The lock is on line 742 `LockRelationOid(ht->main_table_relid, ShareUpdateExclusiveLock);`